lib/Parse/ParseDecl.cpp

```diff
@@ -3347,6 +3349,9 @@ bool Parser::parseDeclList(SourceLoc LBLoc, SourceLoc &RBLoc,
 }
 
 bool Parser::canDelayBodyParsing() {
+  // Calculating interface hash requires tokens consumed in the original order.
+  if (SF.hasInterfaceHash())
+    return false;
```

Comment:
On the one hand this can theoretically be more fine-grained, since you can delay the bodies of functions without affecting the interface hash. But if it's the primary source file and you're doing incremental compilation you're going to need those bodies eventually anyway.

Reply:
the delay of function body parsing isn't controlled by this check and it should be applicable in a broader scenarios. Maybe I should rename this to `canDelayMemberDeclsParsing`.